Treat pseudolog10 and Symlog10 like log10

JamesWrigley opened the pull request:
# Description

The effect is that the tick labels get pretty-printed like with log10. This required a new type union for the regular log functions because they have different limits from pseudolog10/Symlog10.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Added an entry in CHANGELOG.md (for new features and breaking changes)
- [ ] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

JamesWrigley commented:
Looking at the CI history, I think the WGLMakie tests were broken from https://github.com/MakieOrg/Makie.jl/pull/3980.

SimonDanisch commented:
>Looking at the CI history, I think the WGLMakie tests were broken from https://github.com/MakieOrg/Makie.jl/pull/3980.

I updated test images for #4043, you were unlucky to run the ci just after the images where updated and just before rebasing against master ;)

JamesWrigley commented:
Oh dear, I'll rebase again :sweat_smile:

JamesWrigley commented:
(from my side this is ready to go BTW)

SimonDanisch commented:
@jkrumbiegel, does that look correct?

jkrumbiegel commented:
This needs tests / reference tests, ideally exercising these changes for multiple different scenarios where you can use these nonstandard log transforms. Not sure what happens when the functions expecting log ticks encounter 0 or negative values, for example.

JamesWrigley commented:
I added a reference test in 774187496, this is what GLMakie gives me when I run it:
![Non-standard log tick formatting](https://github.com/user-attachments/assets/ee7dae99-3014-4cef-83d1-a67c63b73ecf)

I also added another fix in a2c03f772 to support log formatting ticks are explicitly given, with a reference test:
![Log scale with manual ticks](https://github.com/user-attachments/assets/9241bc94-0653-463a-a24d-87b2f2531985)

JamesWrigley commented:
With luck the unit tests should be fixed in 9976f6574 :crossed_fingers: They were actually testing that the log ticks *weren't* formatted with a log formatting.

JamesWrigley commented:
(bump)

JamesWrigley commented:
(bump)

ffreyer commented on ReferenceTests/src/tests/figures_and_makielayout.jl:
```diff
@@ -133,6 +133,37 @@ end
+    a2.xticks[] = 2 .^ [4, 6]
+    a3.xticks[] = exp.([3, 4])
+    a4.xticks[] = [0, 60, 80, 90, 100]
+    a5.xticks[] = [-10, 0, 10]
```
User supplied ticks with `pseudolog10` are broken here. This range should result in `-10^1, 10^0, 10^1` but generates `10^-1.0413927, 10^0, 10^1.0413927`.

JamesWrigley replied:
Thanks for catching that, I'll try to fix it next week.

jkrumbiegel commented:
Is the refimage from above still current where xs vs xs is plotted? On the left it says 10 ^ -1 while it should be -10 ^ 1
